newIDE/app/src/ResourcesList/ResourceUtils.js

```diff
@@ -34,3 +43,46 @@ export const resourceHasValidPath = (
   const resourcePath = getLocalResourceFullPath(project, resourceName);
   return fs.existsSync(resourcePath);
 };
+
+export const selectLocalResourcePath = (
+  project: gdProject,
+  options: {
+    multiSelections: boolean,
+    title: string,
+    name: string,
+    extensions: Array<string>,
+    forEachPath: any,
+    callback?: ?() => any,
+  }
+) => {
+  return new Promise((resolve, reject) => {
+    if (!dialog) return reject('Not supported');
+
+    const properties = ['openFile'];
+    if (options.multiSelections) properties.push('multiSelections');
+    const projectPath = path.dirname(project.getProjectFile());
+
+    const browserWindow = electron.remote.getCurrentWindow();
+    dialog.showOpenDialog(
+      browserWindow,
+      {
+        title: options.title,
+        properties,
+        filters: [{ name: options.name, extensions: options.extensions }],
+        defaultPath: projectPath,
+      },
+      paths => {
+        if (!paths) return resolve([]);
```

Comment:
Don't have multiple way of getting the result of a function. Promise are already designed to handle this without having to use callbacks. Also avoid doing job in the function with a "forEachXXX" callback, because you're making it harder for the reader to know what's going on ("why passing a forEachXXX? Is there some magic involved in this loop? Just give me the result and I can do a for each by myself.").

Here, you have:
* A promise that return the resources
* A forEachPath that is just a map that you could call yourself and does not provide much value over the promise.
* a callback that does not even have the resources selected, and does not provide much value over the promise.

If you need, read about Promises: https://javascript.info/promise-basics

Just do:
```
dialog.showOpenDialog(...,
  paths => {
    if (!paths) return resolve([]);

    return resolve(paths);
  }
```

and use it like a promise whenever you need it:

```
return selectLocalResourcePath(project, options).then(resources => {
  return resources.map((resourcePath) => {
    // Do whatever you want with your resourcePath, like creating resources:
    const fontResource = new gd.FontResource();
    const projectPath = path.dirname(project.getProjectFile());
    fontResource.setFile(path.relative(projectPath, resourcePath));
    fontResource.setName(path.relative(projectPath, resourcePath));

    return fontResource;
  });
});
```